### PR TITLE
Update Packer module to support GPU and simpler project ID specification

### PR DESCRIPTION
**Pull request by tpdownes**
The Packer module does not presently expose accelerator settings.

This patch:

- ensures the minimum release of the googlecompute plugin that includes the HPC VM Image family by default
- enables access to the configuration parameters for GPU nodes

Without this patch, we cannot provision a machine with accelerators because the default value for `on_host_maintenance` is invalid. Additionally, we cannot use the GCP-supported script for installing GPU drivers because it will quit on machines without GPUs:

```
│ 2022-11-08T23:20:05.700097902Z │ 7133618043896683788 │ startup-script: There doesn't seem to be a GPU unit connected to your system. Aborting drivers installation.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           │
```

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [X] Have you written unit tests to cover this change?
* [ ] Is unit test coverage still above 80%?
* [X ] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: modules/packer/custom-image/variables.pkr.hcl [hunk 1]
```diff
@@ -169,3 +167,25 @@ variable "labels" {
   type        = map(string)
   default     = null
 }
+
+variable "accelerator_type" {
+  description = "Type of accelerator cards to attach to the instance."
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply modules/packer/custom-image/variables.pkr.hcl:172]: No. This does not need to be defined for the a2 series.

I am not entirely sure what happens if you do, but it is likely similar to attempting to attach an accelerator to a family that doesn't support it. I will modify text to make clear that this parameter is unnecessary for A2 series.

File: modules/packer/custom-image/variables.pkr.hcl [hunk 2]
```diff
@@ -169,3 +167,25 @@ variable "labels" {
...
+  type        = string
+  default     = null
+}
+
+variable "accelerator_count" {
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply modules/packer/custom-image/variables.pkr.hcl:177]: No. I did. Unfortunately every time I look there is no packer linter of similar caliber as tflint. I will use it and, upon reflection, set it to null by default as the direct equivalent of the status quo.

**Comment by tpdownes** [comment]
The integration group with this test in it is running.

File: modules/packer/custom-image/variables.pkr.hcl (hunk 2 above)
[the remark this answers is not in the document]
Comment (nick-stroud) [reply modules/packer/custom-image/variables.pkr.hcl:177]: nit: I would also include in the description that it is not needed for A2 shapes.

**Review by nick-stroud — APPROVED** [review]

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**